gcp/gcp.go

```diff
@@ -1127,3 +1144,109 @@ func getDatabaseInstanceStatus(instance *sqladmin.DatabaseInstance) types.State
 		return types.Unknown
 	}
 }
+
+type clusterListAggregator interface {
+	Do(opts ...googleapi.CallOption) (*dataprocpb.ListClustersResponse, error)
+}
+
+func getClusters(listClusters *dataproc.ClusterIterator, region string) ([]*types.Cluster, error) {
+	clusters := make([]*types.Cluster, 0)
+	for {
+		cluster, err := listClusters.Next()
+		if err != nil && err == iterator.Done {
+			break
+		} else if err != nil {
+			log.Debugf("[GET_CLUSTERS] Error fetching cluster list: %+v", err)
+			return nil, err
+		}
+		clusters = append(clusters, newCluster(cluster, region))
+	}
+	return clusters, nil
+}
+
+func newCluster(cluster *dataprocpb.Cluster, region string) *types.Cluster {
+	log.Debugf("[GET_CLUSTERS] %+s in %s has status: %+s", cluster.ClusterName, region, cluster.Status.State)
+	return &types.Cluster{
+		Name:      cluster.ClusterName,
+		Uuid:      cluster.ClusterUuid,
+		Created:   cluster.Status.StateStartTime.AsTime(),
+		CloudType: types.GCP,
+		Region:    region,
+		Tags:      convertTags(cluster.Labels),
+		Config:    cluster.GetConfig(),
+		State:     getClusterState(cluster.Status.GetState()),
+	}
+}
+
+func getClusterState(s dataprocpb.ClusterStatus_State) types.State {
+	// convert ClusterSTatus_State to types.State
+
+	// This array must match the const ordering from here:
+	// https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/dataproc/v1#ClusterStatus_State
+	statuses := []types.State{
```

Comment:
Could you please write a test which guarantees the right order of the array? Or maybe a `map[int32]types.State` should be better option to pair statuses.

```
status, ok := statuses[s]
if !ok return types.Unknown
return status
```